### PR TITLE
WIP: Vulkan/adreno: Apply workaround for Harvest Moon issue #10421

### PR DESCRIPTION
**Pull request by hrydgard**
Props to Unknown for the workaround idea.

### CODE REVIEW

**Comment by hrydgard** [comment]
Yeah, good point @unknownbrackets . Marked it as WIP for now, to think more about, but I believe this can get close to a perfect workaround for this particular bug.. if we do understand it (that is, if we can get it to repro consistently...)

**Comment by zminhquanz** [comment]
And prevent a crash on Persona 3 (deadlock on Adreno driver )

File: GPU/Vulkan/StateMappingVulkan.cpp [hunk 1]
```diff
@@ -235,6 +235,18 @@ void DrawEngineVulkan::ConvertStateToVulkanKey(FramebufferManagerVulkan &fbManag
 			}
 
 			key.colorWriteMask = (rmask ? VK_COLOR_COMPONENT_R_BIT : 0) | (gmask ? VK_COLOR_COMPONENT_G_BIT : 0) | (bmask ? VK_COLOR_COMPONENT_B_BIT : 0) | (amask ? VK_COLOR_COMPONENT_A_BIT : 0);
+
+			// Workaround proposed in #10421
+			if ((gstate.pmskc & 0x00FFFFFF) == 0x00FFFFFF && vulkan_->GetPhysicalDeviceProperties(vulkan_->GetCurrentPhysicalDevice()).vendorID == VULKAN_VENDOR_QUALCOMM) {
+				key.colorWriteMask = VK_COLOR_COMPONENT_R_BIT | VK_COLOR_COMPONENT_G_BIT | VK_COLOR_COMPONENT_B_BIT | VK_COLOR_COMPONENT_A_BIT;
+				key.blendEnable = true;
```
Comment (unknownbrackets) [inline GPU/Vulkan/StateMappingVulkan.cpp]: I think the right workaround would be something like:

```c++
if (!key.blendEnable) {
	key.blendEnable = true;
	key.blendOpAlpha = VK_BLEND_OP_ADD;
	key.srcAlpha = VK_BLEND_FACTOR_ZERO;
	key.destAlpha = VK_BLEND_FACTOR_ONE;
}
key.blendOpColor = VK_BLEND_OP_ADD;
key.srcColor = VK_BLEND_FACTOR_ZERO;
key.destColor = VK_BLEND_FACTOR_ONE;
```

That should avoid breaking any stencil emulation if in place.

-[Unknown]

**Comment by unknownbrackets** [comment]
This Vulkan code shouldn't be related to the Persona 3 GLES bug - is there even another Vulkan bug in Persona 3?

-[Unknown]

File: GPU/Vulkan/StateMappingVulkan.cpp (hunk 1 above)
Comment (hrydgard) [reply GPU/Vulkan/StateMappingVulkan.cpp]: Yeah, I'll change it to that.

**Comment by zminhquanz** [comment]
@unknownbrackets No . Just OpenGL bug on Adreno driver

**Comment by unknownbrackets** [comment]
Seems to work well on my device, and I can't think of any dangers now.

-[Unknown]

**Pull request merged**